homeassistant/components/brunt/__init__.py

```diff
@@ -54,6 +54,7 @@ async def async_update_data():
             if err.status == 401:
                 _LOGGER.warning("Device not found, will reload Brunt integration")
                 await hass.config_entries.async_reload(entry.entry_id)
+            raise UpdateFailed from err
```

Comment:
Added a default `raise UpdateFailed` here to not have to deal with optional data. The alternative would probably be to add asserts in multiple places.